Feature/update dependencies

perruche opened the pull request:
Update out of date dependencies :
- on the project 
- on the template
- depth dependencies to resolve npm vulnerabilities

titouanmathis committed:
Merge tag '1.0.0-beta.0' into develop

v1.0.0-beta.0

Changed
- Use the `meta` CLI to build CSS and JS assets (52c679d, #13)
- Improve the default assets configuration: structure, tools, etc. (e22b27a)

Added
- Add plugins to SAO prompt (dfa8a60, #7)
- Add GitLab CI pipelines (2ac588e, d4d7f7c)
- Add an `acf-json` folder when selecting ACF as a plugin (b5dd81f)
- Add the Studiometa\WP\Assets utility (d923229)
- Add a default value to limit the number of posts revisions (0667b4f)

Fixed
- Fix template errors (bce4924, #14)

Security
- Update NPM and Composer dependencies

titouanmathis committed:
Merge tag '1.0.0-beta.1' into develop

v1.0.0-beta.1

Added
- Add an example .htaccess file (0f6396d)

Changed
- Improve the cache for the GitLab CI pipelines (7390a03)
- Use the full URL in the description instead of the name (1001b1b)
- Update the JS Toolkit to the latest version (9aa37d4)
- Improve the template's readme (d825eda)
- Lint theme PHP files (1ed5763)

Fixed
- Fix a missing comma in composer.json (8cb1884)
- Fix post content render (2ab213e)

titouanmathis committed:
Merge tag '1.0.0-beta.2' into develop

v1.0.0-beta.2

Fixed
- Fix the 4th step of the prompt (58eb5b4)

perruche committed:
Update project dependencies

perruche committed:
Update template dependencies

perruche committed:
Update .eslintrc

titouanmathis commented:
I have a similar feature waiting for commits on my computer, I will update this feature.

perruche committed:
Update project dev dependencies

- Update tailwind.config to use the new preset feature

perruche committed:
Update js files to comply with @studiometa/js-toolkit-beta.1

titouanmathis commented on template/package.json:
```diff
@@ -6,30 +6,29 @@
     "lint": "npm run lint:scripts && npm run lint:styles",
-    "lint:scripts": "eslint web/wp-content/themes/<%= slug %>/src/js --ext=.js,.vue",
-    "lint:styles": "stylelint 'web/wp-content/themes/<%= slug %>/src/**/*.(scss|vue)'"
+    "lint:scripts": "eslint web/wp-content/themes/test2/src/js --ext=.js,.vue",
```
We need the `<%= slug %>` placeholder here.

titouanmathis commented on template/package.json:
```diff
@@ -6,30 +6,29 @@
-    "@babel/preset-env": "^7.10.2",
+    "@babel/plugin-transform-runtime": "^7.12.1",
+    "@babel/preset-env": "^7.12.1",
     "@fullhuman/postcss-purgecss": "^2.2.0",
```
We can delete the Purge CSS package, it is included in the Webpack configuration now.

perruche committed:
Add a blank dist/app.css

To avoid PHP errors on install when no build task have been run. Cause
the file is enqueue by assets.yaml

titouanmathis commented on template/package.json:
```diff
@@ -6,30 +6,29 @@
     "@studiometa/tailwind-config": "^1.0.2",
-    "@studiometa/webpack-config": "^2.0.1",
+    "@studiometa/webpack-config": "^2.4.1",
     "autoprefixer": "^9.8.0",
```
Same as Purge CSS, the `autoprefixer` and `cssnano` packages are included in the Webpack configuration, we can remove them here.

titouanmathis commented on template/package.json:
```diff
@@ -6,30 +6,29 @@
   },
   "dependencies": {
-    "@studiometa/js-toolkit": "^1.0.0-alpha.12"
+    "@studiometa/js-toolkit": "^1.0.0-beta.1"
```
Maybe we can set the version to `latest` to make sure we always have the latest version installed when creating a new project:

```json
{
  "dependencies": {
    "@studiometa/js-toolkit": "latest"
  }
}
```

titouanmathis reviewed (changes requested):
Some minor changes to add, and we can remove the `template/package-lock.json`, it will be generated when creating a new project.

perruche committed:
Add the presets option from @studiometa/webpack-config

- Remove postcss.config.js (included via
@studiometa/webpack-config)

perruche committed:
Remove outdated npm dependencies

perruche committed:
Fix typing errors in package.json

perruche committed:
Use latest version of @studiometa/js-toolkit as default

perruche committed:
Remove template/package-lock.json

titouanmathis commented on .eslintrc.js:
```diff
@@ -1,12 +1,12 @@
-};
\ No newline at end of file
+  extends: '@studiometa/eslint-config',
+  overrides: [
```
We can remove the overrides, the `gulpfile.js` and `webpack.config.js` files are not in the project anymore.

titouanmathis commented on template/tailwind.config.js:
```diff
@@ -1,13 +1,11 @@
  */
-module.exports = merge(config, {
+module.exports = {
+  presets: [require('tailwindcss/defaultConfig'), require('@studiometa/tailwind-config')],
```
👍

perruche committed:
Suppress eslintrc overrides
